runtime/src/lib.rs

```diff
@@ -0,0 +1,263 @@
+			panic!("Runtime is not initialized in Tokio mode.")
+		}
+	}
+
+	/// Returns runtime executor.
+	pub fn executor(&self) -> Executor {
+		self.executor.clone()
+	}
+}
+
+#[derive(Clone)]
+enum Mode {
+	Tokio(TaskExecutor),
+	// Mode used in tests
+	#[allow(dead_code)]
+	Sync,
+	// Mode used in tests
+	#[allow(dead_code)]
+	ThreadPerFuture,
+}
+
+impl fmt::Debug for Mode {
+	fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+		use self::Mode::*;
+
+		match *self {
+			Tokio(_) => write!(fmt, "tokio"),
+			Sync => write!(fmt, "synchronous"),
+			ThreadPerFuture => write!(fmt, "thread per future"),
+		}
+	}
+}
+
+/// Returns a future which runs `f` until `duration` has elapsed, at which
+/// time `on_timeout` is run and the future resolves.
+fn timeout<F, R, T>(f: F, duration: Duration, on_timeout: T) -> impl Future<Item = (), Error = ()> + Send + 'static
+where
+	T: FnOnce() -> () + Send + 'static,
+	F: FnOnce() -> R + Send + 'static,
+	R: IntoFuture<Item = (), Error = ()> + Send + 'static,
+	R::Future: Send + 'static,
+{
+	let future = future::lazy(f);
+	let timeout = Delay::new(Instant::now() + duration).then(move |_| {
+		on_timeout();
+		Ok(())
+	});
+	future.select(timeout).then(|_| Ok(()))
+}
+
+#[derive(Debug, Clone)]
+pub struct Executor {
+	inner: Mode,
+}
+
+impl Executor {
+	/// Executor for existing runtime.
+	#[deprecated(note = "Exists only to connect with current JSONRPC implementation")]
+	pub fn new(executor: TaskExecutor) -> Self {
+		Executor { inner: Mode::Tokio(executor) }
+	}
+
+	/// Synchronous executor, used for tests.
+	#[cfg(any(test, feature = "test-helpers"))]
+	pub fn new_sync() -> Self {
+		Executor { inner: Mode::Sync }
+	}
+
+	/// Spawns a new thread for each future (use only for tests).
+	#[cfg(any(test, feature = "test-helpers"))]
+	pub fn new_thread_per_future() -> Self {
+		Executor { inner: Mode::ThreadPerFuture }
+	}
+
+	/// Spawn a future on this runtime
+	pub fn spawn<R>(&self, r: R)
+	where
+		R: IntoFuture<Item = (), Error = ()> + Send + 'static,
+		R::Future: Send + 'static,
+	{
+		match self.inner {
+			Mode::Tokio(ref executor) => executor.spawn(r.into_future()),
+			Mode::Sync => {
+				let _ = r.into_future().wait();
+			}
+			Mode::ThreadPerFuture => {
+				thread::spawn(move || {
+					let _ = r.into_future().wait();
+				});
+			}
+		}
+	}
+
+	/// Spawn a new future returned by the given closure.
+	pub fn spawn_fn<F, R>(&self, f: F)
+	where
+		F: FnOnce() -> R + Send + 'static,
+		R: IntoFuture<Item = (), Error = ()> + Send + 'static,
+		R::Future: Send + 'static,
+	{
+		match self.inner {
+			Mode::Tokio(ref executor) => executor.spawn(future::lazy(f)),
+			Mode::Sync => {
+				let _ = future::lazy(f).wait();
+			}
+			Mode::ThreadPerFuture => {
+				thread::spawn(move || {
+					let _ = f().into_future().wait();
+				});
+			}
+		}
+	}
+
+	/// Spawn a new future and wait for it or for a timeout to occur.
+	pub fn spawn_with_timeout<F, R, T>(&self, f: F, duration: Duration, on_timeout: T)
```

Comment:
Seems unused. Can you double-check if there is any code using it and delete it if not?